File: app/src/main/java/com/ytempest/wanandroid/interactor/impl/HttpHelperImpl.java
package com.ytempest.wanandroid.interactor.impl;

import com.ytempest.wanandroid.http.HttpApi;
import com.ytempest.wanandroid.http.bean.ArchitectureContentBean;
import com.ytempest.wanandroid.http.bean.ArticleCollectBean;
import com.ytempest.wanandroid.http.bean.BannerBean;
import com.ytempest.wanandroid.http.bean.BaseResp;
import com.ytempest.wanandroid.http.bean.HomeArticleBean;
import com.ytempest.wanandroid.http.bean.KnowledgeArchitectureBean;
import com.ytempest.wanandroid.http.bean.LoginBean;
import com.ytempest.wanandroid.http.bean.MyCollectionBean;
import com.ytempest.wanandroid.http.bean.NavigationListBean;
import com.ytempest.wanandroid.http.bean.OutsideArticleCollectBean;
import com.ytempest.wanandroid.http.bean.ProjectClassifyBean;
import com.ytempest.wanandroid.http.bean.ProjectContentBean;
import com.ytempest.wanandroid.interactor.HttpHelper;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Observable;

/**
 * @author heqidu
 * @since 2020/6/28
 */
public class HttpHelperImpl implements HttpHelper {

    private final HttpApi mHttpApi;

    @Inject
    public HttpHelperImpl(HttpApi httpApi) {
        mHttpApi = httpApi;
    }

    @Override
    public Observable<BaseResp<HomeArticleBean>> getHomeArticleList(int pageNum) {
        return mHttpApi.getHomeArticleList(pageNum);
    }

    @Override
    public Observable<BaseResp<LoginBean>> login(String account, String password) {
        return mHttpApi.login(account, password);
    }

    @Override
    public Observable<BaseResp<LoginBean>> register(String account, String pwd, String confirmPwd) {
        return mHttpApi.register(account, pwd, confirmPwd);
    }

    @Override
    public Observable<BaseResp<List<BannerBean>>> getBannerList() {
        return mHttpApi.getBannerList();
    }

    @Override
    public Observable<BaseResp<MyCollectionBean>> getMyCollectionList() {
        return mHttpApi.getMyCollectionList();
    }

    @Override
    public Observable<BaseResp<ArticleCollectBean>> addCollectArticle(long articleId) {
        return mHttpApi.addCollectArticle(articleId);
    }

    @Override
    public Observable<BaseResp<OutsideArticleCollectBean>> addCollectOutsideArticle(String title, String author, String link) {
        return mHttpApi.addCollectOutsideArticle(title, author, link);
    }

    @Override
    public Observable<BaseResp<ArticleCollectBean>> cancelCollectArticle(long articleId) {
        return mHttpApi.cancelCollectArticle(articleId);
    }

    @Override
    public Observable<BaseResp<ArticleCollectBean>> cancelMyCollectArticle(long articleId) {
        return mHttpApi.cancelMyCollectArticle(articleId);
    }

    @Override
    public Observable<BaseResp<List<KnowledgeArchitectureBean>>> getKnowledgeArchitecture() {
        return mHttpApi.getKnowledgeArchitecture();
    }

    @Override
    public Observable<BaseResp<ArchitectureContentBean>> getArchitectureContent(int id, int page) {
        return mHttpApi.getArchitectureContent(page, id);
    }

    @Override
    public Observable<BaseResp<List<NavigationListBean>>> getNavigationList() {
        return mHttpApi.getNavigationList();
    }

    @Override
    public Observable<BaseResp<List<ProjectClassifyBean>>> getProjectClassify() {
        return mHttpApi.getProjectClassify();
    }

    @Override
    public Observable<BaseResp<ProjectContentBean>> getProjectContent(int page, int cid) {
        return mHttpApi.getProjectContent(page, cid);
    }
}

File: app/src/main/java/com/ytempest/wanandroid/interactor/impl/HttpHelperImpl.kt
package com.ytempest.wanandroid.interactor.impl

import com.ytempest.wanandroid.di.module.http.HttpModule
import com.ytempest.wanandroid.http.HttpApi
import com.ytempest.wanandroid.http.bean.*
import com.ytempest.wanandroid.interactor.HttpHelper
import io.reactivex.Observable
import retrofit2.Call

/**
 * @author heqidu
 * @since 21-2-8
 */

class HttpHelperImpl : HttpHelper {

    private val mHttpApi: HttpApi = HttpModule.instance.getHttpApi()

    override fun getHomeArticleList(pageNum: Int): Call<BaseResp<HomeArticleBean>> =
            mHttpApi.getHomeArticleList(pageNum)

    override fun login(account: String, password: String): Call<BaseResp<LoginBean>> =
            mHttpApi.login(account, password)


    override fun register(account: String?, pwd: String?, confirmPwd: String?): Call<BaseResp<LoginBean>> =
            mHttpApi.register(account, pwd, confirmPwd)

    override fun getBannerList(): Call<BaseResp<List<BannerBean>>> =
            mHttpApi.getBannerList()

    override fun getMyCollectionList(): Observable<BaseResp<MyCollectionBean>> =
            mHttpApi.getMyCollectionList();

    override fun addCollectArticle(articleId: Long): Call<BaseResp<ArticleCollectBean>> =
            mHttpApi.addCollectArticle(articleId);


    override fun addCollectOutsideArticle(title: String?, author: String?, link: String?): Call<BaseResp<OutsideArticleCollectBean>> {
        return mHttpApi.addCollectOutsideArticle(title, author, link)
    }

    override fun cancelCollectArticle(articleId: Long): Call<BaseResp<ArticleCollectBean>> =
            mHttpApi.cancelCollectArticle(articleId)

    override fun cancelMyCollectArticle(articleId: Long): Observable<BaseResp<ArticleCollectBean>> =
            mHttpApi.cancelMyCollectArticle(articleId)

    override fun getKnowledgeArchitecture(): Call<BaseResp<List<KnowledgeArchitectureBean>>> =
            mHttpApi.getKnowledgeArchitecture()

    override fun getArchitectureContent(id: Int, page: Int): Call<BaseResp<ArchitectureContentBean>> {
        return mHttpApi.getArchitectureContent(page, id)
    }

    override fun getNavigationList(): Call<BaseResp<List<NavigationListBean>>> =
            mHttpApi.getNavigationList()

    override fun getProjectClassify(): Call<BaseResp<List<ProjectClassifyBean>>> =
            mHttpApi.getProjectClassify()

    override fun getProjectContent(page: Int, cid: Int): Call<BaseResp<ProjectContentBean>> =
            mHttpApi.getProjectContent(page, cid)
}